packages/flow-core/src/components/f-tag/f-tag.ts

```diff
@@ -308,9 +319,36 @@ export class FTag extends FRoot {
 			?selected=${this.selected}
 			?clickable=${this.clickable}
 		>
-			${iconLeft}${this.label}${counter}${iconRight}
+			${iconLeft}
+			<f-div class="text-content">${this.label}</f-div>
+			${counter}${iconRight}
 		</div>`;
 	}
+
+	protected async updated(
+		changedProperties: PropertyValueMap<any> | Map<PropertyKey, unknown>
+	): Promise<void> {
+		super.updated(changedProperties);
+		requestAnimationFrame(() => {
```

Comment:
requestAnimationFrame will execute on every re-paint of the browser, remove it.

just check the character count of `this.label` in a render function and add tooltip attribute based on that